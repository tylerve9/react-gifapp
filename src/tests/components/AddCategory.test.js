import React from 'react';
import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('Pruebas para el <AddCategory /> ', () => {
    
    //const setCategories = () => {};
    const setCategories = jest.fn();

    let wrapper = shallow( <AddCategory setCategories={ setCategories } /> );

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={ setCategories } /> );
    });

    test('Debe mostrar el componente correctamente ', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe de cambiar la caja de texto ', () => {
        let inputText = wrapper.find('input');
        const value = 'Miguel';
        inputText.simulate('change', { target: {
            value
        }});

        //expect( inputText.prop('value') ).toBe( value );
    });
    
    test('No debe de postear la caja del submit', () => {
        wrapper.find('form').simulate('submit', { preventDefault(){} });
        expect( setCategories ).not.toHaveBeenCalled();
    });

    test('Debe de postear la caja del submit', () => {
        const inputText = wrapper.find('input');
        const value = 'Miguel';
        inputText.simulate('change', { target: {
            value
        }});
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect( setCategories ).toHaveBeenCalled();
        expect( setCategories ).toHaveBeenCalledTimes(1);
        expect( setCategories ).toHaveBeenCalledWith( expect.any(Function) );
        expect( inputText.prop('value') ).toBe('');
    });
    

    
});
