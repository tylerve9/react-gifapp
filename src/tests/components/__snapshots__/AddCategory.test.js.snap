// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para el <AddCategory />  Debe mostrar el componente correctamente  1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Search..."
    type="text"
    value=""
  />
</form>
`;
